### PR TITLE
Add omrsysinfo_get_processor_feature_name on x86

### PR DESCRIPTION
**Pull request by harryyu1994**
Given a feature in uint32_t, the api returns its corresponding feature name.

Signed-off-by: Harry Yu <harryyu1994@gmail.com>

### CODE REVIEW

**Comment by harryyu1994** [comment]
@rwy0717 Hi Robert, could you help review this one? Thanks!

File: port/common/omrsysinfo_helpers.c [hunk 1]
```diff
@@ -92,6 +92,145 @@
...
+	"sha",              /* 3 * 32 + 29 */
+	"avx512bw",         /* 3 * 32 + 30 */
+	"avx512vl"          /* 3 * 32 + 31 */
+};
+
+const char *
+omrsysinfo_get_x86_processor_feature_name(uint32_t feature)
+{
+	if (feature < 0 || feature >= sizeof(OMR_FEATURE_X86_NAME)/sizeof(const char *))
+		return "null";
```
Comment (rwy0717) [inline port/common/omrsysinfo_helpers.c:230]: needs {braces} around the body of the if.

File: port/win32/omrsysinfo.c [hunk 2]
```diff
@@ -146,6 +146,16 @@ omrsysinfo_processor_set_feature(struct OMRPortLibrary *portLibrary, OMRProcesso
 	return rc;
 }
 
+const char*
+omrsysinfo_get_processor_feature_name(struct OMRPortLibrary *portLibrary, uint32_t feature)
+{
+	const char* rc = "null";
+	Trc_PRT_sysinfo_get_processor_feature_name_Entered(feature);
+	rc = omrsysinfo_get_x86_processor_feature_name(feature);;
```
Comment (rwy0717) [inline port/win32/omrsysinfo.c:154]: Extra semicolon.

File: port/common/omrsysinfo_helpers.c [hunk 3]
```diff
@@ -92,6 +92,147 @@
...
+	"avx512vl"          /* 3 * 32 + 31 */
+};
+
+const char *
+omrsysinfo_get_x86_processor_feature_name(uint32_t feature)
+{
+	if (feature < 0 || feature >= sizeof(OMR_FEATURE_X86_NAME)/sizeof(const char *))
+	{
+		return "null";
+	}
```
Comment (rwy0717) [inline port/common/omrsysinfo_helpers.c:232]:
```suggestion
	if (feature < 0 || feature >= sizeof(OMR_FEATURE_X86_NAME)/sizeof(const char *)) {
		return "null";
	}
```

**Comment by rwy0717** [comment]
@genie-omr build all

**Comment by rwy0717** [comment]
```
[2020-07-21T17:22:40.781Z] /Users/jenkins/workspace/Build/port/common/omrsysinfo_helpers.c:229:14: error: comparison of unsigned expression < 0 is always false [-Werror,-Wtautological-compare]
[2020-07-21T17:22:40.781Z]         if (feature < 0 || feature >= sizeof(OMR_FEATURE_X86_NAME)/sizeof(const char *)) {
[2020-07-21T17:22:40.781Z]             ~~~~~~~ ^ ~
[2020-07-21T17:22:40.781Z] 1 error generated.
```

**Comment by rwy0717** [comment]
@genie-omr build all

File: port/common/omrport.tdf [hunk 4]
```diff
@@ -1114,6 +1114,9 @@ TraceExit=Trc_PRT_sysinfo_processor_disable_feature_Exit Group=sysinfo Overhead=
 TraceEntry=Trc_PRT_sysinfo_processor_set_feature_Entered Group=sysinfo Overhead=1 Level=5 NoEnv Template="sysinfo_processor_set_feature: desc = %p, feature = %d, enable = %d"
 TraceExit=Trc_PRT_sysinfo_processor_set_feature_Exit Group=sysinfo Overhead=1 Level=5 NoEnv Template="sysinfo_processor_set_feature: returning with %zd"
 
+TraceEntry=Trc_PRT_sysinfo_get_processor_feature_name_Entered Group=sysinfo Overhead=1 Level=5 NoEnv Template="sysinfo_get_processor_feature_name: feature = %d"
+TraceExit=Trc_PRT_sysinfo_get_processor_feature_name_Exit Group=sysinfo Overhead=1 Level=5 NoEnv Template="sysinfo_get_processor_feature_name: returning with %s"
+
```
Comment (keithc-ca) [inline port/common/omrport.tdf:1119]: New tracepoints should be added to the end of the file so existing tracepoint numbers remain stable.
See https://github.com/eclipse/openj9/blob/master/doc/diagnostics/AddingTracepoints.md.

File: port/common/omrsysinfo.c [hunk 5]
```diff
@@ -146,6 +146,23 @@ omrsysinfo_processor_set_feature(struct OMRPortLibrary *portLibrary, OMRProcesso
...
+ * @param[in] portLibrary The port library.
+ * @param[in] feature The processor feature. (see omrport.h for list of features OMRPORT_FEATURE_{PPC,S390,PPC}_*)
+ *
+ * @return string literals of the given processor feature
+ */
+const char*
+omrsysinfo_get_processor_feature_name(struct OMRPortLibrary *portLibrary, uint32_t feature)
+{
+	Trc_PRT_sysinfo_get_processor_feature_name_Entered(feature);
+	const char* rc = "null";
```
Comment (keithc-ca) [inline port/common/omrsysinfo.c:161]: This is not legal C code: the declaration must be at the beginning of a block.

File: port/common/omrport.tdf (hunk 4 above)
Comment (harryyu1994) [reply port/common/omrport.tdf:1119]: Is it too late now to open a new PR to change this?

Comment (harryyu1994) [reply port/common/omrport.tdf:1119]: made a new PR here: https://github.com/eclipse/omr/pull/5422

**Pull request merged**